File: day-02/Go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

const (
	Keyword      = "Nemo"
	NotFoundText = "I can't find Nemo :("
	FoundText    = "I found Nemo at %v!\n"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Println("Type the Nemo sentence:")

	input, err := reader.ReadString('\n')

	fmt.Println("........Looking for Nemo..........")

	if err != nil {
		fmt.Printf("Something went wrong: %v\n", err)
	} else {
		nemoPosition := findNemo(input)

		if nemoPosition > 0 {
			fmt.Printf(FoundText, nemoPosition)
		} else {
			fmt.Printf(NotFoundText)
		}
	}

}

func findNemo(input string) int {
	for i, word := range strings.Fields(input) {

		if word == Keyword {
			return i + 1
		}
	}
	return 0
}

File: day-02/c++/main.cpp
#include <iostream>

int main(int argc, char *argv[])
{       
    bool found;    
    for (int i = 1; i < argc; i++) {        
        found = std::string(argv[i]).compare("Nemo") == 0;
        if (found) {
            std::cout << "I found Nemo at " << i << "!" << std::endl;
            break;
        }
    }

    if (!found) {
        std::cout << " I can't find Nemo :(" << std::endl;
    }

    return 0;
}